instances/models.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package instances
+
+import "sdk.kraft.cloud/services"
+
+// CreateRequest is the payload for a POST /instances request.
+// https://docs.kraft.cloud/api/v1/instances/#creating-a-new-instance
+type CreateRequest struct {
+	Name          string                     `json:"name"`
+	Image         string                     `json:"image"`
+	Args          []string                   `json:"args"`
+	Env           map[string]string          `json:"env"`
+	MemoryMB      int                        `json:"memory_mb"`
+	ServiceGroup  *CreateRequestServiceGroup `json:"service_group"`
+	Volumes       []CreateRequestVolume      `json:"volumes"`
+	Autostart     bool                       `json:"autostart"`
+	Replicas      int                        `json:"replicas"`
+	WaitTimeoutMs int                        `json:"wait_timeout_ms"`
+	Features      []Feature                  `json:"features"`
```

Comment:
But I see also some requests don't have omitted fields.


Reminder that if `ServiceGroup`  is not completely missing, a default one will be created

Is this intentional?

Reply:
Inside a request, a Go `nil` value translates to a JSON `null`, which is valid for the KraftCloud API. Since the consumers of the SDK never see the generated JSON payload, is there any benefit to make _request_ fields omitempty? (for responses, this is a different story)

Reply:
We'll have to see inside kraftkit once it's testable.

I mentioned `ServiceGroup` because that had the bug/feture, that if left empty (so an existing requests, but with empty fields), UKP will generate a default service group for it. This is now a pointer so should be fine.

Reply:
I just double checked: `service_group: null` and no `service_group` at all are handled the same way by the API (as expected).